[CMSP-481] Remove `upstream-require` script

jazzsequence opened the pull request:
This PR removes the `UpstreamRequire` method in `upstream-configuration/scripts/ComposerScripts.php` and adds the Composer hijacking magic ✨ that removes it from customer `composer.json` files without us creating merge conflicts.

**Note:** Before this upstream is officially GA, we should actually remove the `require` and `scripts-description` for `upstream-require` since, at that point, people should have already have updated and not have it anymore OR be brand new and should not be given something that's going to immediately be removed.

jazzsequence committed:
add suggest to root composer.json

jazzsequence committed:
remove UpstreamRequire

jazzsequence committed:
remove updatelocaldependencies & add magic to unset upstream-require

jazzsequence committed:
fix linting issues

jazzsequence committed:
remove scripts-descriptions if empty

kporras07 commented on upstream-configuration/scripts/ComposerScripts.php:
```diff
@@ -97,34 +42,157 @@ public static function preUpdate(Event $event)
+        }
+
+        // Remove our upstream convenience scripts, if the user has not removed them.
+        if (isset($composerJson['scripts']['upstream-require'])) {
```
This will be running these changes every time someone runs `composer update` (which includes `composer require`). Are you ok with this?

In DCM, we added a new post-update-hook and made sure to run this code only if that hook was absent: https://github.com/pantheon-systems/drupal-composer-managed/blob/default/upstream-configuration/scripts/ComposerScripts.php#L108

jazzsequence replied:
Makes sense. Updated to include the same.

kporras07 commented on upstream-configuration/scripts/ComposerScripts.php:
```diff
@@ -97,34 +42,157 @@ public static function preUpdate(Event $event)
+     * @param Composer\Script\Event $event
+     *   The Event object passed in from Composer
+     */
+    public static function postUpdate(Event $event)
```
I think you're not using this anywhere

kporras07 reviewed (commented):
One question and one other comment.

Also, maybe we should add some docs about the composer-package? In DCM we added a small README: https://github.com/pantheon-systems/drupal-composer-managed/blob/default/upstream-configuration/README.md#upstream-configuration (it's ok if you consider that the composer suggestion is enough)

jazzsequence committed:
add our postUpdate hook & remove upstream-require if it doesn't exist

jazzsequence commented:
> Also, maybe we should add some docs about the composer-package?

We have that, too: https://github.com/pantheon-systems/wordpress-composer-managed/blob/default/upstream-configuration/README.md

jazzsequence committed:
Apply --dry-run fixes

copy pasta from https://github.com/pantheon-upstreams/wordpress-composer-managed/pull/4/files

jazzsequence committed:
closing brace on new line because psr-4 is dumb

kporras07 reviewed (approved):
LGTM
